src/lib/cache.ts

```diff
@@ -28,24 +28,30 @@ export const cacheAsync = <T>(
             return Promise.resolve(cache[key] as T);
         } else {
             // First read attempt on this key. Fetch an initial value and setup the refresh scheduler
-            const result: T = await fn();
-            cache[key] = result;
+            try {
+                const result: T = await fn();
+                cache[key] = result;
+                return Promise.resolve(result);
+            } catch (err) {
+                console.log(`Failed to make initial request for ${key}: ${err}`);
+                return Promise.reject(
+                    new Error(`Failed to make initial request for ${key}: ${err}`),
+                );
+            } finally {
+                const scheduleRefresh = (ms: number): void => {
```

Comment:
Maybe this could be rewritten as a setInterval with a clearInterval, rather than a mind-melding callback, wrapped in an async await, wrapped in a recursive call?!

Reply:
`setInterval` might be cleaner though it still gets fiddly with the error handling + retries
I will revisit this later